notification.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Notification struct {
+	Type     string
+	Instance string
+	Status   NotificationStatus
+}
+
+type NotificationStatus string
+
+const (
+	NotificationMaster NotificationStatus = "MASTER"
+	NotificationFault                     = "FAULT"
+	NotificationBackup                    = "BACKUP"
+)
```

Comment:
I wanted to double-check that this works how I thought it works (i.e. all consts get type `NotificationStatus`), but with my simple test program which does `fmt.Printf("%T %T %T", NotificationMaster, NotificationFault, NotificationBackup)` I get `main.NotificationStatus string string`, and not the expected 3x `main.NotificationStatus`. Notably, a lot of documentation online still uses the form you've used here, so I'm now really unsure whether we should have explicit type `NotificationStatus` for all consts here.

Reply:
You're right. These should also be explicitly typed. I think I made that mistake before.  I somehow though the `iota` syntactic sugar applied here too, which isn't the case

Without it they are untyped constant, which means they can be used as both a `string` and a `NotificationStatus`, so it worked in all cases here, but it's better to be explicit here.